x-pack/plugins/security_solution/common/api/detection_engine/rule_monitoring/rule_execution_logs/get_rule_execution_results/get_rule_execution_results_route.gen.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { z } from 'zod';
+
+/*
+ * NOTICE: Do not edit this file manually.
+ * This file is automatically generated by the OpenAPI Generator, @kbn/openapi-generator.
+ */
+
+import { RuleExecutionStatus } from '../../model/execution_status.gen';
+import {
+  SortFieldOfRuleExecutionResult,
+  RuleExecutionResult,
+} from '../../model/execution_result.gen';
+import { SortOrder } from '../../../model/sorting.gen';
+
+export type GetRuleExecutionResultsRequestQuery = z.infer<
+  typeof GetRuleExecutionResultsRequestQuery
+>;
+export const GetRuleExecutionResultsRequestQuery = z.object({
+  /**
+   * Start date of the time range to query
+   */
+  start: z.string().datetime(),
+  /**
+   * End date of the time range to query
+   */
+  end: z.string().datetime(),
+  /**
+   * Query text to filter results by
+   */
+  query_text: z.string().optional().default(''),
+  /**
+   * Comma-separated list of rule execution statuses to filter results by
+   */
+  status_filters: z
+    .preprocess(
+      (value: unknown) =>
+        typeof value === 'string' ? (value === '' ? [] : value.split(',')) : value,
+      z.array(RuleExecutionStatus)
+    )
+    .optional()
+    .default([]),
+  /**
+   * Field to sort results by
+   */
+  sort_field: SortFieldOfRuleExecutionResult.optional().default('timestamp'),
+  /**
+   * Sort order to sort results by
+   */
+  sort_order: SortOrder.optional().default('desc'),
+  /**
+   * Page number to return
+   */
+  page: z.number().int().optional().default(1),
+  /**
+   * Number of results per page
+   */
+  per_page: z.number().int().optional().default(20),
+});
```

Comment:
Looks like these aren't making across the wire btw:

<img width="1360" alt="image" src="https://github.com/elastic/kibana/assets/2946766/75f08f26-6c04-4327-8099-dc40fdabdd5c">

Reply:
Thanks for catching that 👍 The schema didn't accept string numbers, such as `"20"`. So, I modified the query string schema to `z.coerce.number()`, which now accepts strings too. I've also added an extra test case to cover that logic.